s2wav/filters.py

```diff
@@ -23,9 +25,10 @@ def filters_axisym(L: int, lam: float, J_min: int) -> Tuple[np.ndarray, np.ndarr
     Args:
         L (int): Harmonic band-limit.
 
-        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        J_min (int, optional): Lowest frequency wavelet scale to be used. Defaults to 0.
 
-        J_min (int): First wavelet scale used.
+        lam (float, optional): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+            Note that :math:`\lambda = 2` indicates dyadic wavelets. Defaults to 2.
 
     Raises:
         ValueError: L is not an integer.
```

Comment:
This is a TypeError

Reply:
Same with J_min. Because we use them for range() they need to be of type int. Could change this to TypeError or consider allowing floats and then converting them to integers but then we'd need to check if input float is mathematically an integer.

Reply:
I just meant that in this function we use the values of L and J_min to define iteration limits in range(). So even if the value is eg 0.0, the code will break. This could cause potential issues if someone developing the code was for instance reading in values from a file and they were defined in Python as floats even though they are mathematically integers. This error might be more helpful in a function that is explicitly for the users (not sure which ones we plan those to be) but it wouldn't hurt to have these checks to make life easier for people contributing to the code so that if something goes wrong we get and explanation that's already interpreted in terms of our algorithm instead of having to figure it out themselves from the Python errors. I would just leave it as a type error though. 

That was my thinking but we do have type hints for J_min and L which convey they should be ints. I'm not sure what the common practice is regarding this, might be a bit over the top to have these checks.

Reply:
Hmm, I think that we just have to draw a line somewhere. The overall synthesis and analysis (and their adjoint) transforms are the main functions exposed to the user, so if we were going to add any TypeErrors they would be there, but for now I think we just remove these TypeErrors and once the package is complete we could consider adding some simple function to assert all variables are of the correct type, but really users should handle this on their end and check the api docs to see what is expected

Reply:
Sounds good!